app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  before_action :require_user, only: [:destroy]
+
+  def new
+    redirect_to home_path unless !logged_in?
+  end
+  def create
+    user = User.find_by(email: params[:email])
+
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      flash[:success] = "Welcome again #{user.full_name}"
+      redirect_to home_path
+    else
+      flash[:danger] = "Invalid user or password.Please try again"

```

Comment:
When you render - always think about flash that you have to use only for 1 current request. `flash.now` so it does not carry to next request.